books/spiders/books_spider.py

```diff
@@ -0,0 +1,29 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books_spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs):
+        book_list = response.css(".product_pod")
+        for book in book_list:
+            url = book.css("h3 a::attr(href)").get()
+            yield response.follow(url, self.parse_single_book)
+
+        next_page = response.urljoin(response.css(".next a::attr(href)").get())
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
```

Comment:
The `next_page` logic assumes that the `next_page` URL is always present. However, if there is no next page, `response.css(".next a::attr(href)").get()` will return `None`, and `response.urljoin(None)` will result in the current page URL. Consider checking if `next_page` is not `None` before calling `response.urljoin` to avoid unnecessary requests.